tests/unit/magic_strings.py

```diff
@@ -0,0 +1,95 @@
+from pytest import raises
```

Comment:
I’d love to see this file become an integration test, so that we can also cover the `@magic_string` mechanism. Can be a follow-up PR, though. 👍